patch/patcher.go

```diff
@@ -531,12 +532,29 @@ func (p *Patcher) patchIdent(id *ast.Ident, obj types.Object) {
 			if v.Tag == nil {
 				v.Tag = &ast.BasicLit{}
 			}
-			v.Tag.Value = "`" + strings.TrimSpace(strings.Trim(v.Tag.Value, "` ")+" "+tags) + "`"
+
+			v.Tag.Value = "`" + mergeTags(v.Tag.Value, tags) + "`"
 			log.Printf("Add tags:\t%q.%s %s", obj.Pkg().Path(), id.Name, v.Tag.Value)
 		}
 	}
 }
 
+// Override tags
+func mergeTags(oldTag, newTag string) string {
+	oldTags := strings.Split(strings.TrimSpace(strings.Trim(oldTag, "` ")), " ")
+	var newTags []string
+	for _, tag := range oldTags {
+		var key string
+		if kv := strings.Split(tag, ":"); len(kv) > 0 {
+			key = kv[0]
+		}
+		if value, exist := reflect.StructTag(newTag).Lookup(key); !exist || value == "" {
+			newTags = append(newTags, tag)
+		}
+	}
+	return strings.Join(append(newTags, newTag), " ")
```

Comment:
Instead of using `reflect.StructTag` and concatenating strings, I think it might be better to use a `map` to record the old tags, then merge / overwrite the new tags in the map, then join into a string as a final step.

Ideally it also has a string slice to maintain the order of the tags.

This would have the additional property of being able to override existing tags with an empty value.